packages/block-editor/src/components/off-canvas-editor/index.js

```diff
@@ -208,9 +211,48 @@ function __ExperimentalOffCanvasEditor(
 						shouldShowInnerBlocks={ shouldShowInnerBlocks }
 						selectBlockInCanvas={ selectBlockInCanvas }
 					/>
+					<tr>
+						<td>
+							<OffCanvasEditorAppender
+								rootClientId={ clientId }
+							/>
+						</td>
+					</tr>
```

Comment:
Any focusable element should be wrapped in `<TreeGridRow>` and `<TreeGridCell>` instead of `<tr>` and `<td>` so that keyboard navigation works.

I see `ListView` has `AnimatedTreeGridRow`, not sure if it needs to use that. I think that's for the block movement animation.

There may be a few extra requirements like increasing the `setSize` prop (the number of items in a branch of the tree) on the `TreeGridRow` by one when there's an appender.

Also `TreeGridCell` has callback style children, and the focusable should be able to receive the `ref`, `tabIndex` and `onFocus` props.

Reply:
TIL about these components auto managing focus. Thanks! Will update.

Reply:
@talldan I updated things. Does that look better? 

I haven't dealt with setSize. I suppose that's when the appender is hidden/shown we're going to need to dynamically update that?

Reply:
`TreeGridRow` does take some props:
- level: the level of nesting
- position: the row number
- setSize: the number of rows

They correspond to the aria props (aria-level, aria-setsize, aria-posinset), more info here - https://www.w3.org/WAI/ARIA/apg/example-index/treegrid/treegrid-1.

Something that I didn't notice until now is that I think it would be easier to move the code to the `TreeGridBranch` component, as it already has those variables ready for you to pass as props (`level`, `position`, `blockCount`).

At the moment, the way it works in this PR is that the appender only shows at root level, if you want to keep that behavior you can do conditional rendering (when `level === 1`), but moving the implementation to `TreeGridBranch` would also allow you to render an appender at every level (it would be visually very busy, so possibly something to iterate on).

May also need to double-check the keyboard navigation still works as expected as I remember last time the movers were made visible it was buggy. I can definitely help with that.

Reply:
Nice. I didn't know about those aria props. I'll take another look at this now.

Reply:
> At the moment, the way it works in this PR is that the appender only shows at root level, if you want to keep that behavior you can do conditional rendering (when level === 1), but moving the implementation to TreeGridBranch would also allow you to render an appender at every depth (it would be visually very busy, so possibly something to iterate on).

I tried this out. The problem was with `level` set to `1` the appender showed after every root level block. This isn't what we want _yet_ so I'd prefer to avoid overcomplicating this PR with any additional logic or changes required to achieve that,

I agree it's something we _may_ want in the future and we can look to refactor towards that when that becomes a requirement. 

I appreciate all your feedback on this aspect. It's been extremely helpful 🙇

Reply:
I added all the required props and hard coded them to reflect the root level.